pkg/client.go

```diff
@@ -45,8 +45,7 @@ func (client *ClickHouseClient) Query(query string) (*Response, error) {
 
 	if client.settings.Instance.BasicAuthEnabled {
 		password, _ := client.settings.Instance.DecryptedSecureJSONData["basicAuthPassword"]
-		req.Header.Set("X-ClickHouse-User", client.settings.Instance.BasicAuthUser)
```

Comment:
I think we can add X-ClickHouse-* headers and basic auth both for this corner case

clickhouse itself support both methods of authorization  in same request

but if you use some third party API gateway, it may require basic authorization

so, let stay Headers.Set and SetBasicAuth together in this case